### PR TITLE
Add workflow for releasing post install hook

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

This PR adds a manual workflow which will build and push the new helm post install hook.

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: .github/workflows/release-post-install-hook.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          version: ${{ github.event.inputs.version }}
+        run: |
+          echo "Release ${version} already exists"
+          exit 1
+
+      - name: Push Post Install Hook To Registry
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (leo-ri) [inline .github/workflows/release-post-install-hook.yml:53]: secrets.QUAY_PASSWORD (?)

File: .github/workflows/release-post-install-hook.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        env:
+          version: ${{ github.event.inputs.version }}
+        run: |
+          echo "Release ${version} already exists"
+          exit 1
+
+      - name: Push Post Install Hook To Registry
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (leo-ri) [inline .github/workflows/release-post-install-hook.yml:52]: probably QUAY_ROBOT_NAME

Comment (chatton) [reply .github/workflows/release-post-install-hook.yml:52]: I don't see this username in our secrets 🤔

Comment (chatton) [reply .github/workflows/release-post-install-hook.yml:52]: ah I see you mean from same workflow, not secrets!

**Review by leo-ri — APPROVED** [review]

**Pull request merged**